### ISSUE TITLE
Compilation error in src/view/surface.cpp

### ISSUE THREAD

**Issue by ericonr** [WayfireWM/wayfire#432]
I tested this with both clang and gcc, both generated errors. This is the error output from gcc:

```
[72/160] Compiling C++ object 'src/25a6634@@wayfire@exe/view_surface.cpp.o'.
FAILED: src/25a6634@@wayfire@exe/view_surface.cpp.o
c++ -Isrc/25a6634@@wayfire@exe -Isrc -I../src -I. -I../ -I../src/api -I/usr/lib/libffi-3.2.1/include -I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/libevdev-1.0/ -I/usr/include/libxml2 -I/usr/include/libpng16 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -std=c++17 -g -DWLR_USE_UNSTABLE -Wno-unused-parameter -DHAS_ADDR2LINE=1 -MD -MQ 'src/25a6634@@wayfire@exe/view_surface.cpp.o' -MF 'src/25a6634@@wayfire@exe/view_surface.cpp.o.d' -o 'src/25a6634@@wayfire@exe/view_surface.cpp.o' -c ../src/view/surface.cpp
../src/view/surface.cpp: In member function ‘virtual wlr_buffer* wf::wlr_surface_base_t::get_buffer()’:
../src/view/surface.cpp:325:25: error: cannot convert ‘wlr_client_buffer*’ to ‘wlr_buffer*’ in return
  325 |         return surface->buffer;
      |                ~~~~~~~~~^~~~~~
      |                         |
      |                         wlr_client_buffer*
```

Misc:

* GCC: c++ (Arch Linux 9.2.1+20200130-2) 9.2.1 20200130
* git commit: `1b87bff15`

**Comment by ericonr** [comment]
So, this seems to be due to a wlroots update. They changed the structure of the wlr_surface struct. I've found a fix, I'm going to try it out now.

**Comment by ammen99** [comment]
Fixed in #433
